### DISCUSSION TITLE
No model with the specified URI or the matching Input/Output type is found.

### DISCUSSION THREAD

**Discussion by yangming-wang** [deepjavalibrary/djl#3405] [Q&A]
![微信图片_20240811183756](https://github.com/user-attachments/assets/4ded6aa2-9a86-4def-8b88-ea5fff95000a)
Is there any expert who has encountered this problem before? Please help me with the solution.

**Comment by frankfliu** [comment]
@yangming-wang 

You need specify an `TranslatorFactory`. If you download the model from DJL model zoo, our model zoo metadata already defined a `TranslatorFactory`, see: https://github.com/deepjavalibrary/djl/blob/master/engines/pytorch/pytorch-model-zoo/src/test/resources/mlrepo/model/nlp/question_answer/ai/djl/pytorch/bertqa/metadata.json#L28

When you load the model from the local file, you can define `TranslatorFactory` in two ways:
1. use `optTranslatorFactory(new PtBertQATranslatorFactory())
2. create a `serving.properties` file and `translatorFactory=ai.djl.pytorch.zoo.nlp.qa.PtBertQATranslatorFactory"

By the way, we recommend you use Huggingface model, it's faster and standardize. See example: https://github.com/deepjavalibrary/djl-demo/blob/master/huggingface/nlp/src/main/java/com/examples/QuestionAnswering.java

**Reply by yangming-wang** [reply]
Thank you very much, this is very helpful to me！
